### PR TITLE
PC-1294: Update Django version

### PR DESCRIPTION
**Pull request by samyou-softwire**
[Link to Jira ticket](https://beisdigital.atlassian.net/browse/PC-1294)

# Description

updates django to 5.1 & django-gov-notify to 0.5.0

makes upgrades, both mentioned in https://docs.djangoproject.com/en/5.1/releases/5.0/#features-removed-in-5-0:
- `USE_L10N` is no longer used, so is removed
- `pytz` package is no longer used, preferring python core packages

have tried frontdoor and portal, no issues found

# Checklist

- [x] I have made any necessary updates to the documentation
- [x] I have checked there are no unnecessary IDE warnings in this PR
- [x] I have checked there are no unintentional line ending changes
- [x] I have added tests where applicable
- [x] If I have made any changes to the code, I have used the IDE auto-formatter on it
- [x] If I have made any changes to Python files (even if not changing any content strings), I have run `make extract-translations`

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -20,7 +20,7 @@ testino = "0.3.9"
 optional = true
 
 [tool.poetry.group.deployment.dependencies]
-django-gov-notify = "0.4.0"
+django-gov-notify = "0.5.0"
```
Comment (jdgage) [inline pyproject.toml:23]: There's a v0.6.0 as well - can we go to that?

File: pyproject.toml [hunk 2]
```diff
@@ -33,7 +33,7 @@ python = "~3.11.9"
 boto3 = "1.26.86"
 botocore = "1.29.86"
 debugpy = "1.6.7"
-Django = "3.2.19"
+Django = "5.1.1"
```
Comment (jdgage) [inline pyproject.toml:36]: 5.1.5 is the latest by the looks of things: would prefer to go to the latest patch

File: pyproject.toml [hunk 3]
```diff
@@ -33,7 +33,7 @@ python = "~3.11.9"
...
 django-allauth = "0.54.0"
 django-cors-headers = "4.0.0"
 django-csp = "3.7"
```
Comment (jdgage) [inline pyproject.toml:39]: Would be worth checking notes for all of these other Django-related packages to see if they support Django 5.x.

We're probably best off bumping all of them up to vLatest tbh

**Review by jdgage — CHANGES_REQUESTED** [review]
Is there a ticket for bumping the Python version further? That can be unblocked once this is in.

File: pyproject.toml (hunk 2 above)
Comment (samyou-softwire) [reply pyproject.toml:36]: I could only find [5.1.4](https://pypi.org/project/Django/5.1.4/)

File: pyproject.toml (hunk 1 above)
Comment (samyou-softwire) [reply pyproject.toml:23]: sure

File: pyproject.toml (hunk 3 above)
Comment (samyou-softwire) [reply pyproject.toml]: have done most of these, though to note:
- the update for `django-allauth` seems to be breaking, would need some more research into how we interact with this library
- `django-use-email-as-username` is now [no longer actively maintained](https://github.com/jmfederico/django-use-email-as-username)

Comment (samyou-softwire) [reply pyproject.toml]: agreed in catchup to ensure `django-allauth` is compatible with django 5.0. backlog ticket to be raised for `django-use-email-as-username`

Comment (samyou-softwire) [reply pyproject.toml]: have updated `django-allauth` to a version known to work with django 5.0. won't go any further as there are many breaking changes to consider https://docs.allauth.org/en/latest/release-notes/recent.html

**Review by jdgage — APPROVED** [review]
Looks good: checking we've got tickets for any follow-up upgrades, as discussed earlier?

File: pyproject.toml (hunk 3 above)
Comment (samyou-softwire) [reply pyproject.toml]: have added [PC-1634](https://beisdigital.atlassian.net/browse/PC-1634), [PC-1635](https://beisdigital.atlassian.net/browse/PC-1635) & [PC-1636](https://beisdigital.atlassian.net/browse/PC-1636) to cover removing use email as username & implementing a package auditor across our services

**Pull request merged**